File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def count_divisor(n):
    cnt = 0
    for i in range(1, int(n ** 0.5) + 1):
        if n % i == 0:
            cnt += 1 if i == n // i else 2
    return cnt
    
def solution(number, limit, power):
    result = 0
    for i in range(1, number + 1):
        divisor = count_divisor(i)
        result += power if divisor > limit else divisor
    return result

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.js
function countDivisor(n){
    let cnt = 0;
    for(let i = 1 ; i <= Math.sqrt(n) ; i++){
        if(n % i === 0) cnt += (i === n / i) ? 1 : 2;  // 제곱수는 중복 방지
    }
    return cnt;
}

function solution(number, limit, power) {
    let result = 0;
    for(let i = 1 ; i <= +number ; i++){
        const divisor = countDivisor(i);
        result += divisor > limit ? power : divisor;
    }
    return result;
}